## Marketing Data Analysis
### Complete SQL Analysis Notebook

This notebook contains a comprehensive marketing data analysis including:
- Data quality assessment and cleaning
- Descriptive statistics
- Customer segmentation and behavior analysis
- Campaign performance evaluation
- Product and channel performance analysis

## 0.0 Set Up Table

In [ ]:
CREATE TABLE marketing_data (
    ID                  INT,
    Year_Birth          INT,
    Education           VARCHAR(50),
    Marital_Status      VARCHAR(50),
    Income              DECIMAL(12, 2),
    Kidhome             TINYINT,
    Teenhome            TINYINT,
    Dt_Customer         DATE,
    Recency             INT,
    MntWines            INT,
    MntFruits           INT,
    MntMeatProducts     INT,
    MntFishProducts     INT,
    MntSweetProducts    INT,
    MntGoldProds        INT,
    NumDealsPurchases   INT,
    NumWebPurchases     INT,
    NumCatalogPurchases INT,
    NumStorePurchases   INT,
    NumWebVisitsMonth   INT,
    AcceptedCmp3        TINYINT,
    AcceptedCmp4        TINYINT,
    AcceptedCmp5        TINYINT,
    AcceptedCmp1        TINYINT,
    AcceptedCmp2        TINYINT,
    Response            TINYINT,
    Complain            TINYINT,
    Country             VARCHAR(50)
);

In [ ]:
BULK INSERT marketing_data
FROM 'D:\My Project Datasets\EDA\Marketing+Data\marketing_data.csv'
WITH (
    FORMAT = 'CSV',
    FIRSTROW = 2,
    FIELDTERMINATOR = ',',
    ROWTERMINATOR = '\n',
    TABLOCK
);


## 1. Data Quality Assessment

### 1.1 Initial Data Exploration

In [89]:
SELECT TOP 10 *
FROM marketing_data;

(10 rows affected)

ID    | Year_Birth | Education  | Marital_Status | Income   | Kidhome | Teenhome | Dt_Customer | Recency | MntWines | MntFruits | MntMeatProducts | MntFishProducts | MntSweetProducts | MntGoldProds | NumDealsPurchases | NumWebPurchases | NumCatalogPurchases | NumStorePurchases | NumWebVisitsMonth | AcceptedCmp3 | AcceptedCmp4 | AcceptedCmp5 | AcceptedCmp1 | AcceptedCmp2 | Response | Complain | Country  
------+------------+------------+----------------+----------+---------+----------+-------------+---------+----------+-----------+-----------------+-----------------+------------------+--------------+-------------------+-----------------+---------------------+-------------------+-------------------+--------------+--------------+--------------+--------------+--------------+----------+----------+----------
1826  | 1970       | Graduation | Divorced       | 84835.00 | 0       | 0        | 2014-06-16  | 0       | 189      | 104       | 379             | 111             | 

<b>

### 1.2 Row Count & Duplicate Check

In [90]:
-- Total records count
SELECT COUNT(*) AS TotalRows
FROM marketing_data;

-- Duplicate ID check
SELECT
    COUNT(ID) AS TotalIDs,
    COUNT(DISTINCT ID) AS UniqueIDs,
    COUNT(ID) - COUNT(DISTINCT ID) AS DuplicateCount
FROM marketing_data;

-- Show duplicate IDs if any
SELECT ID, COUNT(*) AS Occurrences
FROM marketing_data
GROUP BY ID
HAVING COUNT(*) > 1
ORDER BY Occurrences DESC;

(1 row affected)
(1 row affected)
(0 rows affected)

TotalRows
---------
2240     
(1 row)

TotalIDs | UniqueIDs | DuplicateCount
---------+-----------+---------------
2240     | 2240      | 0             
(1 row)

(0 rows)

Total execution time: 00:00:00.073

<b>
The dataset contains 2,240 rows. No duplicates were found, confirming a clean 1-to-1 relationship between each customer and each row.
<b>

<b>

### 1.3 NULL Value Audit (All Columns)

In [ ]:
-- Check for NULL values across all columns

SELECT
    SUM(CASE WHEN ID                  IS NULL THEN 1 ELSE 0 END) AS Null_ID,
    SUM(CASE WHEN Year_Birth          IS NULL THEN 1 ELSE 0 END) AS Null_Year_Birth,
    SUM(CASE WHEN Education           IS NULL THEN 1 ELSE 0 END) AS Null_Education,
    SUM(CASE WHEN Marital_Status      IS NULL THEN 1 ELSE 0 END) AS Null_MaritalStatus,
    SUM(CASE WHEN Income              IS NULL THEN 1 ELSE 0 END) AS Null_Income,
    SUM(CASE WHEN Kidhome             IS NULL THEN 1 ELSE 0 END) AS Null_Kidhome,
    SUM(CASE WHEN Teenhome            IS NULL THEN 1 ELSE 0 END) AS Null_Teenhome,
    SUM(CASE WHEN Dt_Customer         IS NULL THEN 1 ELSE 0 END) AS Null_DtCustomer,
    SUM(CASE WHEN Recency             IS NULL THEN 1 ELSE 0 END) AS Null_Recency,
    SUM(CASE WHEN MntWines            IS NULL THEN 1 ELSE 0 END) AS Null_MntWines,
    SUM(CASE WHEN MntFruits           IS NULL THEN 1 ELSE 0 END) AS Null_MntFruits,
    SUM(CASE WHEN MntMeatProducts     IS NULL THEN 1 ELSE 0 END) AS Null_MntMeat,
    SUM(CASE WHEN MntFishProducts     IS NULL THEN 1 ELSE 0 END) AS Null_MntFish,
    SUM(CASE WHEN MntSweetProducts    IS NULL THEN 1 ELSE 0 END) AS Null_MntSweet,
    SUM(CASE WHEN MntGoldProds        IS NULL THEN 1 ELSE 0 END) AS Null_MntGold,
    SUM(CASE WHEN NumDealsPurchases   IS NULL THEN 1 ELSE 0 END) AS Null_NumDeals,
    SUM(CASE WHEN NumWebPurchases     IS NULL THEN 1 ELSE 0 END) AS Null_NumWeb,
    SUM(CASE WHEN NumCatalogPurchases IS NULL THEN 1 ELSE 0 END) AS Null_NumCatalog,
    SUM(CASE WHEN NumStorePurchases   IS NULL THEN 1 ELSE 0 END) AS Null_NumStore,
    SUM(CASE WHEN NumWebVisitsMonth   IS NULL THEN 1 ELSE 0 END) AS Null_WebVisits,
    SUM(CASE WHEN Response            IS NULL THEN 1 ELSE 0 END) AS Null_Response,
    SUM(CASE WHEN Complain            IS NULL THEN 1 ELSE 0 END) AS Null_Complain,
    SUM(CASE WHEN Country             IS NULL THEN 1 ELSE 0 END) AS Null_Country
FROM marketing_data;

(1 row affected)

Null_ID | Null_Year_Birth | Null_Education | Null_MaritalStatus | Null_Income | Null_Kidhome | Null_Teenhome | Null_DtCustomer | Null_Recency | Null_MntWines | Null_MntFruits | Null_MntMeat | Null_MntFish | Null_MntSweet | Null_MntGold | Null_NumDeals | Null_NumWeb | Null_NumCatalog | Null_NumStore | Null_WebVisits | Null_Response | Null_Complain | Null_Country
--------+-----------------+----------------+--------------------+-------------+--------------+---------------+-----------------+--------------+---------------+----------------+--------------+--------------+---------------+--------------+---------------+-------------+-----------------+---------------+----------------+---------------+---------------+-------------
0       | 0               | 0              | 0                  | 24          | 0            | 0             | 0               | 0            | 0             | 0              | 0            | 0            | 0             | 0            | 0             | 

<b>
Most columns show 0 NULL values, but Income contains 24 missing values (~1.07% of 2,240 records). All other visible columns are complete. Before analysis, the 24 NULL Income records should be handled, either imputed (e.g., using median income) or excluded, depending on the analytical approach.
<b>

<b>

### 1.4 Outlier Detection

#### - Income Outliers (Z-Score > 3)

<b>
Z-Score > 3 = "To investigate records." It does NOT mean "Delete the records." Every flagged record must be reviewed individually.
<b>

In [94]:
-- Detect income outliers using Z-score
WITH IncomeStats AS (
    SELECT
        AVG(Income) AS AvgIncome,
        STDEV(Income) AS StdIncome
    FROM marketing_data
    WHERE Income IS NOT NULL
)
SELECT
    ID,
    Income,
    Year_Birth,
    Education,
    Marital_Status,
    ROUND((Income - s.AvgIncome) / s.StdIncome, 2) AS ZScore
FROM marketing_data m
CROSS JOIN IncomeStats s
WHERE ABS((Income - s.AvgIncome) / s.StdIncome) > 3
  AND Income IS NOT NULL
ORDER BY ZScore DESC;

(8 rows affected)

ID    | Income    | Year_Birth | Education  | Marital_Status | ZScore
------+-----------+------------+------------+----------------+-------
9432  | 666666.00 | 1977       | Graduation | Together       | 24.41 
1503  | 162397.00 | 1976       | PhD        | Together       | 4.38  
1501  | 160803.00 | 1982       | PhD        | Married        | 4.31  
5336  | 157733.00 | 1971       | Master     | Together       | 4.19  
4931  | 157146.00 | 1977       | Graduation | Together       | 4.17  
8475  | 157243.00 | 1973       | PhD        | Married        | 4.17  
11181 | 156924.00 | 1949       | PhD        | Married        | 4.16  
5555  | 153924.00 | 1975       | Graduation | Divorced       | 4.04  
(8 rows)

Total execution time: 00:00:00.311

<b>
Income in the ~ $153K–$162K range is 4 standard deviations above the mean, but still logically acceptable for senior professional income, so these rows were retained. Conversely, income values ​​exceeding $162K have no plausible business explanation and are likely typos, so these rows were removed from the dataset.
<b>

In [131]:
-- Delete rows with unrealistic income (above the reasonable limit of $162K)
DELETE FROM marketing_data
WHERE Income > 162397;

(1 row affected)

Total execution time: 00:00:00.529

#### - Age Outliers (Year_Birth)

In [132]:
-- Detect age outliers
SELECT
    ID,
    Year_Birth,
    2015 - Year_Birth AS Age_In_2015,
    Income,
    Education
FROM marketing_data
WHERE (2015 - Year_Birth) > 90
   OR (2015 - Year_Birth) < 18
ORDER BY Year_Birth ASC;

(3 rows affected)

ID    | Year_Birth | Age_In_2015 | Income   | Education
------+------------+-------------+----------+----------
11004 | 1893       | 122         | 60182.00 | 2n Cycle 
1150  | 1899       | 116         | 83532.00 | PhD      
7829  | 1900       | 115         | 36640.00 | 2n Cycle 
(3 rows)

Total execution time: 00:00:00.074

<b>
Expected output: 3 records, customers born in 1893 (age 122), 1899 (age 116), and 1900 (age 115). These ages are biologically impossible and are confirmed data entry errors  
<b>

#### Remove Age Outliers

In [133]:
-- Delete age outliers (>90 years old)

DELETE FROM marketing_data
WHERE (2015 - Year_Birth) > 90;

(3 rows affected)

Total execution time: 00:00:02.255

<b>

### 1.5 Descriptive Statistics - Numeric Columns

In [3]:
-- Summary statistics for numeric columns

SELECT
    ROUND(MIN(Income), 0)  AS Income_Min,
    ROUND(MAX(Income), 0)  AS Income_Max,
    CAST(ROUND(AVG(Income), 0) AS DECIMAL(10,2))  AS Income_Avg,
    ROUND(STDEV(Income), 0) AS Income_StdDev,

    MIN(2015 - Year_Birth) AS Age_Min,
    MAX(2015 - Year_Birth) AS Age_Max,
    CAST(AVG(2015.0 - Year_Birth) AS DECIMAL(10,2)) AS Age_Avg,

    MIN(Recency) AS Recency_Min,
    MAX(Recency) AS Recency_Max,
    CAST(AVG(Recency * 1.0) AS DECIMAL(10,2)) AS Recency_Avg,

    CAST(AVG(MntWines        * 1.0) AS DECIMAL(10,2)) AS Wines_Avg,
    CAST(AVG(MntFruits       * 1.0) AS DECIMAL(10,2)) AS Fruits_Avg,
    CAST(AVG(MntMeatProducts * 1.0) AS DECIMAL(10,2)) AS Meat_Avg,
    CAST(AVG(MntFishProducts * 1.0) AS DECIMAL(10,2)) AS Fish_Avg,
    CAST(AVG(MntSweetProducts* 1.0) AS DECIMAL(10,2)) AS Sweet_Avg,
    CAST(AVG(MntGoldProds    * 1.0) AS DECIMAL(10,2)) AS Gold_Avg,

    CAST(AVG(NumWebPurchases     * 1.0) AS DECIMAL(10,2)) AS WebPurch_Avg,
    CAST(AVG(NumCatalogPurchases * 1.0) AS DECIMAL(10,2)) AS CatalogPurch_Avg,
    CAST(AVG(NumStorePurchases   * 1.0) AS DECIMAL(10,2)) AS StorePurch_Avg,
    CAST(AVG(NumWebVisitsMonth   * 1.0) AS DECIMAL(10,2)) AS WebVisits_Avg,
    CAST(AVG(NumDealsPurchases   * 1.0) AS DECIMAL(10,2)) AS Deals_Avg
FROM marketing_data


(1 row affected)

Income_Min | Income_Max | Income_Avg | Income_StdDev | Age_Min | Age_Max | Age_Avg | Recency_Min | Recency_Max | Recency_Avg | Wines_Avg | Fruits_Avg | Meat_Avg | Fish_Avg | Sweet_Avg | Gold_Avg | WebPurch_Avg | CatalogPurch_Avg | StorePurch_Avg | WebVisits_Avg | Deals_Avg
-----------+------------+------------+---------------+---------+---------+---------+-------------+-------------+-------------+-----------+------------+----------+----------+-----------+----------+--------------+------------------+----------------+---------------+----------
1730.00    | 162397.00  | 51959.00   | 21527         | 19      | 75      | 46.10   | 0           | 99          | 49.12       | 304.13    | 26.28      | 166.98   | 37.54    | 27.08     | 43.98    | 4.09         | 2.66             | 5.80           | 5.32          | 2.33     
(1 row)

Total execution time: 00:00:00.075

<b>
Average income is $51,959 with high variance. The average customer is 46 year old. Wines dominate spending at avg $304, followed by Meat ($166) and Gold ($44). Fruits and Sweets are the least purchased. Customers make ~ 5 - 6 store purchases, ~4 web purchases, and ~2 catalog purchases per period. Recency averages ~49 days, suggesting a moderately engaged customer base.
<b>

<b><b>

## 2. Customer Demographics & Behavior

### 2.1 Education Level Distribution

In [ ]:
-- Education distribution

SELECT
    Education,
    COUNT(*) AS CustomerCount,
    CAST(100.0 * COUNT(*) / SUM(COUNT(*)) OVER() AS DECIMAL(10,2)) AS PctOfTotal
FROM marketing_data
GROUP BY Education
ORDER BY CustomerCount DESC;

(5 rows affected)

Education  | CustomerCount | PctOfTotal
-----------+---------------+-----------
Graduation | 1126          | 50.36     
PhD        | 485           | 21.69     
Master     | 370           | 16.55     
2n Cycle   | 201           | 8.99      
Basic      | 54            | 2.42      
(5 rows)

Total execution time: 00:00:00.675

<b>
Graduation-level customers dominate at ~50% (1,126 customers), followed by PhD (~21%) and Master (~16%). This is a highly educated customer base. Basic education accounts for only 2.4% (54 customers) and 2n Cycle = 9 % (201). Marketing messaging can safely use sophisticated language and value-based arguments. Premium product campaigns should target the Graduation, PhD, and Master segments.
<b>

<b>

### 2.2 Marital Status Distribution

In [137]:
-- Marital status distribution with quality flag
SELECT
    Marital_Status,
    COUNT(*) AS CustomerCount,
    CAST(ROUND(100.0 * COUNT(*) / SUM(COUNT(*)) OVER(), 1) AS DECIMAL(5,2)) AS PctOfTotal,
    CASE
        WHEN Marital_Status IN ('YOLO','Absurd','Alone') THEN 'Non-Standard — Review'
        ELSE 'Standard'
    END AS DataQuality
FROM marketing_data
GROUP BY Marital_Status
ORDER BY CustomerCount DESC;

(8 rows affected)

Marital_Status | CustomerCount | PctOfTotal | DataQuality          
---------------+---------------+------------+----------------------
Married        | 864           | 38.60      | Standard             
Together       | 578           | 25.80      | Standard             
Single         | 479           | 21.40      | Standard             
Divorced       | 231           | 10.30      | Standard             
Widow          | 77            | 3.40       | Standard             
Alone          | 3             | 0.10       | Non-Standard — Review
YOLO           | 2             | 0.10       | Non-Standard — Review
Absurd         | 2             | 0.10       | Non-Standard — Review
(8 rows)

Total execution time: 00:01:19.901

<b>
Married (39%) and Together (26%) represent nearly two-thirds of customers, 64.4% live with a partner. Single customers account for 21%, Divorced 10%, and Widow 3.4%. Notably, there are 7 records with non-standard values (Alone: 3, YOLO: 2, Absurd: 2), these are likely data entry errors and should be reclassified as Unknown or excluded from marital-status segmentation.
<b>

<b>

### 2.3 Total & Average Spend by Product Category

In [ ]:
-- Product category spend analysis

WITH Totals AS (
    SELECT
        SUM(MntWines) AS TotalWines,
        SUM(MntFruits) AS TotalFruits,
        SUM(MntMeatProducts) AS TotalMeat,
        SUM(MntFishProducts) AS TotalFish,
        SUM(MntSweetProducts) AS TotalSweet,
        SUM(MntGoldProds) AS TotalGold,
        SUM(MntWines + MntFruits + MntMeatProducts +
            MntFishProducts + MntSweetProducts + MntGoldProds) AS GrandTotal
    FROM marketing_data
)
SELECT
    Category,
    TotalSpend,
    CAST(ROUND(TotalSpend * 100.0 / GrandTotal, 1) AS DECIMAL(10,2)) AS PctOfTotalSpend,
    CAST(ROUND(TotalSpend * 1.0 / 2240, 1) AS DECIMAL(10,2)) AS AvgSpendPerCustomer
FROM (
    SELECT 'Wines' AS Category, TotalWines AS TotalSpend, GrandTotal FROM Totals UNION ALL
    SELECT 'Meat', TotalMeat, GrandTotal FROM Totals UNION ALL
    SELECT 'Gold', TotalGold, GrandTotal FROM Totals UNION ALL
    SELECT 'Fish', TotalFish, GrandTotal FROM Totals UNION ALL
    SELECT 'Sweet', TotalSweet, GrandTotal FROM Totals UNION ALL
    SELECT 'Fruits', TotalFruits, GrandTotal FROM Totals
) sub
ORDER BY TotalSpend DESC;

(6 rows affected)

Category | TotalSpend | PctOfTotalSpend | AvgSpendPerCustomer
---------+------------+-----------------+--------------------
Wines    | 680029     | 50.20           | 303.60             
Meat     | 373375     | 27.60           | 166.70             
Gold     | 98346      | 7.30            | 43.90              
Fish     | 83931      | 6.20            | 37.50              
Sweet    | 60552      | 4.50            | 27.00              
Fruits   | 58753      | 4.30            | 26.20              
(6 rows)

Total execution time: 00:01:08.124

<b>
Wines account for ~50% of all product spending, making them by far the most important revenue category. Meat products are second at ~28%. Together, Wines and Meat represent nearly 78% of all spend. Gold is the third-largest category at 7.3% (avg spend: $43.90 per customer). Fish, Sweets, and Fruits are relatively minor, each below 7%. Business implication: Wine promotions and premium meat bundles offer the highest revenue leverage. Gold products also present an upsell opportunity worth exploring. Fruits and Sweets may need targeted promotions to grow their share.
<b>

<b>

### 2.4 Purchase Volume by Channel

In [139]:
-- Total purchases by channel
SELECT
    'Store' AS Channel, SUM(NumStorePurchases) AS TotalPurchases FROM marketing_data UNION ALL
    SELECT 'Web', SUM(NumWebPurchases) FROM marketing_data UNION ALL
    SELECT 'Catalog', SUM(NumCatalogPurchases) FROM marketing_data UNION ALL
    SELECT 'Deals/Discounts', SUM(NumDealsPurchases) FROM marketing_data
ORDER BY TotalPurchases DESC;

-- Average per customer
SELECT
    CAST(ROUND(AVG(NumStorePurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgStorePurchases,
    CAST(ROUND(AVG(NumWebPurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgWebPurchases,
    CAST(ROUND(AVG(NumCatalogPurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgCatalogPurchases,
    CAST(ROUND(AVG(NumDealsPurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgDealsPurchases,
    CAST(ROUND(AVG(NumWebVisitsMonth * 1.0), 2) AS DECIMAL(10,2)) AS AvgWebVisitsPerMonth
FROM marketing_data;

(4 rows affected)
(1 row affected)

Channel         | TotalPurchases
----------------+---------------
Store           | 12959         
Web             | 9140          
Catalog         | 5955          
Deals/Discounts | 5201          
(4 rows)

AvgStorePurchases | AvgWebPurchases | AvgCatalogPurchases | AvgDealsPurchases | AvgWebVisitsPerMonth
------------------+-----------------+---------------------+-------------------+---------------------
5.80              | 4.09            | 2.66                | 2.33              | 5.32                
(1 row)

Total execution time: 00:00:02.681

<b>
In-store purchases are the dominant channel (~39% of all purchases), followed by Web at ~27%. Catalog purchases, while lower volume (~18%), represent a premium channel often associated with higher-value orders. Deals/Discounts account for ~16%, meaningful volume but at the cost of margins. Notably, avg web visits per month is ~5 but avg web purchases is ~4, suggesting reasonable web conversion but room to improve.
<b>

<b>

### 2.5 Campaign Acceptance Rates

In [140]:
-- Campaign performance comparison
SELECT
    Campaign,
    Accepted,
    TotalCustomers,
    CAST(ROUND(100.0 * Accepted / TotalCustomers, 2) AS DECIMAL(10,2)) AS AcceptanceRate_Pct
FROM (
    SELECT 'Campaign 1' AS Campaign, SUM(AcceptedCmp1) AS Accepted, COUNT(*) AS TotalCustomers FROM marketing_data UNION ALL
    SELECT 'Campaign 2', SUM(AcceptedCmp2), COUNT(*) FROM marketing_data UNION ALL
    SELECT 'Campaign 3', SUM(AcceptedCmp3), COUNT(*) FROM marketing_data UNION ALL
    SELECT 'Campaign 4', SUM(AcceptedCmp4), COUNT(*) FROM marketing_data UNION ALL
    SELECT 'Campaign 5', SUM(AcceptedCmp5), COUNT(*) FROM marketing_data UNION ALL
    SELECT 'Final (Response)', SUM(Response), COUNT(*) FROM marketing_data
) campaigns
ORDER BY AcceptanceRate_Pct DESC;

(6 rows affected)

Campaign         | Accepted | TotalCustomers | AcceptanceRate_Pct
-----------------+----------+----------------+-------------------
Final (Response) | 334      | 2236           | 14.94             
Campaign 4       | 167      | 2236           | 7.47              
Campaign 3       | 163      | 2236           | 7.29              
Campaign 5       | 162      | 2236           | 7.25              
Campaign 1       | 144      | 2236           | 6.44              
Campaign 2       | 30       | 2236           | 1.34              
(6 rows)

Total execution time: 00:00:00.937

<b>
The most recent campaign (Response / Campaign 6) has the highest acceptance rate at 14.9%, suggesting improvement over time. Campaign 2 performed worst at 1.3% — far below the others. Campaigns 3, 4, and 5 are clustered around 7-7.4%, indicating consistent moderate performance. The upward trend culminating in Response suggests the business is learning from past campaigns and better targeting. Campaign 2 should be investigated for what made it fail (wrong segment, channel, or offer).  
<b>

<b><b>

## 3. Segmentation Analysis

### 3.1 Income Bracket vs. Average Total Spend

In [141]:
-- Income-based spending analysis

WITH CustomerSpend AS (
    SELECT Income,
        MntWines+MntFruits+MntMeatProducts+
        MntFishProducts+MntSweetProducts+MntGoldProds AS TotalSpend,
        CASE
            WHEN Income < 20000 THEN '1. < $20K'
            WHEN Income BETWEEN 20000 AND 39999 THEN '2. $20K-$40K'
            WHEN Income BETWEEN 40000 AND 59999 THEN '3. $40K-$60K'
            WHEN Income BETWEEN 60000 AND 79999 THEN '4. $60K-$80K'
            WHEN Income BETWEEN 80000 AND 99999 THEN '5. $80K-$100K'
            WHEN Income >= 100000 THEN '6. $100K+'
        END AS IncomeBracket
    FROM marketing_data
    WHERE Income IS NOT NULL 
)
SELECT IncomeBracket, COUNT(*) AS Customers,
    ROUND(AVG(TotalSpend), 0) AS AvgTotalSpend,
    CAST(ROUND(AVG(Income), 0) AS DECIMAL(10,2)) AS AvgIncome,
    CAST(ROUND(AVG(TotalSpend * 1.0) / NULLIF(AVG(Income * 1.0), 0) * 100, 2) AS DECIMAL(10,2)) AS SpendPctOfIncome
FROM CustomerSpend
WHERE IncomeBracket IS NOT NULL
GROUP BY IncomeBracket
ORDER BY IncomeBracket;

(6 rows affected)

IncomeBracket | Customers | AvgTotalSpend | AvgIncome | SpendPctOfIncome
--------------+-----------+---------------+-----------+-----------------
1. < $20K     | 127       | 72            | 14356.00  | 0.51            
2. $20K-$40K  | 604       | 98            | 31100.00  | 0.32            
3. $40K-$60K  | 642       | 401           | 49826.00  | 0.80            
4. $60K-$80K  | 624       | 1108          | 69509.00  | 1.59            
5. $80K-$100K | 203       | 1551          | 85374.00  | 1.82            
6. $100K+     | 12        | 893           | 136016.00 | 0.66            
(6 rows)

Total execution time: 00:00:44.068

<b>
There is a strong positive relationship between income and total spend up to the $80K–$100K bracket, where average spend peaks at $1,551, approximately 21x more than customers earning below $20K. However, the $100K+ group shows an unexpected drop in both absolute spend ($893) and spend-as-a-percentage-of-income (0.66%), though this bracket contains only 12 customers, making it statistically unreliable. The $80K–$100K bracket also shows the highest income allocation rate (1.82%), suggesting it represents the most engaged, premium-spending segment. Income remains a strong targeting variable, with the $80K–$100K bracket as the primary focus for premium campaigns.
<b>

<b>

### 3.2 Education vs. Spend & Campaign Response

In [142]:
-- Education level impact on spending and campaigns

WITH CustomerProfile AS (
    SELECT Education,
        MntWines+MntFruits+MntMeatProducts+
        MntFishProducts+MntSweetProducts+MntGoldProds AS TotalSpend,
        AcceptedCmp1+AcceptedCmp2+AcceptedCmp3+
        AcceptedCmp4+AcceptedCmp5+Response AS TotalCampaignsAccepted,
        Income
    FROM marketing_data
    WHERE Income IS NOT NULL 
)
SELECT Education, COUNT(*) AS Customers,
    CAST(ROUND(AVG(Income), 0) AS DECIMAL(10,2)) AS AvgIncome,
    CAST(ROUND(AVG(TotalSpend), 0) AS DECIMAL(10,2)) AS AvgTotalSpend,
    CAST(ROUND(AVG(TotalCampaignsAccepted * 1.0), 2) AS DECIMAL(10,2)) AS AvgCampaignsAccepted,
    CAST(ROUND(100.0 * SUM(CASE WHEN TotalCampaignsAccepted > 0 THEN 1 ELSE 0 END) / COUNT(*), 1) AS DECIMAL(10,2)) AS PctResponded_AnyCampaign
FROM CustomerProfile
GROUP BY Education
ORDER BY AvgTotalSpend DESC;

(5 rows affected)

Education  | Customers | AvgIncome | AvgTotalSpend | AvgCampaignsAccepted | PctResponded_AnyCampaign
-----------+-----------+-----------+---------------+----------------------+-------------------------
PhD        | 480       | 56088.00  | 674.00        | 0.55                 | 32.70                   
Graduation | 1115      | 52170.00  | 622.00        | 0.44                 | 26.50                   
Master     | 365       | 52918.00  | 609.00        | 0.43                 | 27.10                   
2n Cycle   | 198       | 47625.00  | 499.00        | 0.36                 | 22.70                   
Basic      | 54        | 20306.00  | 81.00         | 0.15                 | 13.00                   
(5 rows)

Total execution time: 00:00:58.254

<b>
PhD customers lead across all metrics, highest average spend ($674), highest income ($56K), and highest campaign response rate (32.7%). Graduation-level customers are the largest segment (1,115 customers) and rank second in average spend ($622), outspending Master-level customers ($609), making them a high-volume, moderate-to-high value segment. Basic education customers spend the least ($81) and respond to campaigns at the lowest rate (13%). Strategy: Allocate premium campaign budget toward the PhD segment for quality, leverage Graduation-level customers for volume with value-focused messaging, and deprioritize Basic-education customers in campaign spend.
<b>

<b>

### 3.3 Children at Home vs. Spending & Channel

In [143]:
-- Impact of children on spending behavior

SELECT
    Kidhome + Teenhome AS TotalChildren,
    COUNT(*) AS Customers,
    CAST(ROUND(AVG(MntWines * 1.0), 0)          AS DECIMAL(10,2)) AS AvgWineSpend,
    CAST(ROUND(AVG(MntMeatProducts * 1.0), 0)   AS DECIMAL(10,2)) AS AvgMeatSpend,
    CAST(ROUND(AVG(MntFruits * 1.0), 0)         AS DECIMAL(10,2)) AS AvgFruitSpend,
    CAST(ROUND(AVG(MntSweetProducts * 1.0), 0)  AS DECIMAL(10,2)) AS AvgSweetSpend,

    CAST(ROUND(AVG(MntWines + MntFruits + MntMeatProducts +
        MntFishProducts + MntSweetProducts + MntGoldProds) * 1.0, 0) AS DECIMAL(10,2)) AS AvgTotalSpend,

    CAST(ROUND(AVG(NumWebPurchases * 1.0), 2)   AS DECIMAL(10,2)) AS AvgWebPurchases,
    CAST(ROUND(AVG(NumStorePurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgStorePurchases,
    CAST(ROUND(AVG(NumDealsPurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgDealsPurchases,

    CAST(ROUND(100.0 * AVG(CAST(Response AS FLOAT)), 1) AS DECIMAL(10,2)) AS LastCampaign_AcceptRate
FROM marketing_data
GROUP BY Kidhome + Teenhome
ORDER BY TotalChildren;

(4 rows affected)

TotalChildren | Customers | AvgWineSpend | AvgMeatSpend | AvgFruitSpend | AvgSweetSpend | AvgTotalSpend | AvgWebPurchases | AvgStorePurchases | AvgDealsPurchases | LastCampaign_AcceptRate
--------------+-----------+--------------+--------------+---------------+---------------+---------------+-----------------+-------------------+-------------------+------------------------
0             | 637       | 487.00       | 372.00       | 52.00         | 53.00         | 1104.00       | 4.39            | 7.27              | 1.14              | 26.50                  
1             | 1125      | 268.00       | 99.00        | 19.00         | 20.00         | 473.00        | 4.29            | 5.60              | 2.47              | 10.30                  
2             | 421       | 142.00       | 51.00        | 8.00          | 8.00          | 245.00        | 3.23            | 4.31              | 3.56              | 11.20                  
3             | 53        | 171.00       

<b>
Child-free customers: highest spend ($1,104), highest store purchases (7.27), highest campaign acceptance (26.5%). Deals purchases increase with number of children (1.14 → 3.79). Store purchases DECREASE as number of children increases.
<b>

<b>

### 3.4 Recency Band vs. Spend & Purchases

In [ ]:
-- Customer recency segmentation

WITH RecencyBands AS (
    SELECT *,
        CASE
            WHEN Recency <= 25 THEN '1. 0-25 days (Very Recent)'
            WHEN Recency <= 50 THEN '2. 26-50 days'
            WHEN Recency <= 75 THEN '3. 51-75 days'
            ELSE '4. 76+ days (Dormant)'
        END AS RecencyBand,

        MntWines + MntFruits + MntMeatProducts +
        MntFishProducts + MntSweetProducts + MntGoldProds AS TotalSpend,

        NumWebPurchases + NumCatalogPurchases + NumStorePurchases AS TotalPurchases

    FROM marketing_data
)
SELECT

    RecencyBand,
    COUNT(*) AS Customers,
    CAST(ROUND(AVG(TotalSpend), 0) AS DECIMAL(10,2)) AS AvgTotalSpend,
    CAST(ROUND(AVG(TotalPurchases * 1.0), 2) AS DECIMAL(10,2)) AS AvgPurchases,
    CAST(ROUND(100.0 * AVG(CAST(Response AS FLOAT)), 1) AS DECIMAL(10,2)) AS LastCampaign_AcceptRate

FROM RecencyBands
GROUP BY RecencyBand
ORDER BY RecencyBand;

(4 rows affected)

RecencyBand                | Customers | AvgTotalSpend | AvgPurchases | LastCampaign_AcceptRate
---------------------------+-----------+---------------+--------------+------------------------
1. 0-25 days (Very Recent) | 590       | 584.00        | 12.53        | 25.90                  
2. 26-50 days              | 551       | 587.00        | 12.33        | 14.00                  
3. 51-75 days              | 562       | 650.00        | 12.90        | 11.70                  
4. 76+ days (Dormant)      | 533       | 601.00        | 12.42        | 7.10                   
(4 rows)

Total execution time: 00:00:50.878

<b>
Recency is a strong signal for campaign response rate, which clearly declines from 25.9% (0–25 days) to 7.1% (76+ days). However, spend and purchase frequency do not follow the same pattern, the 51–75 day band shows the highest spend ($650) and purchase frequency (12.90), suggesting these customers are high-value but less campaign-responsive. Dormant customers (76+ days) still spend $601 on average, indicating they have not disengaged from buying entirely. Strategy: Prioritize the 0–25 day segment for campaign activation (highest response rate), use re-engagement offers for the 51–75 day band to convert their high spend potential, and apply win-back messaging for 76+ day dormant customers.
<b>

<b>

### 3.5 Web Visits vs. Web Purchases

In [145]:
-- Web behavior analysis

SELECT
    NumWebVisitsMonth AS WebVisits,
    COUNT(*) AS Customers,
    CASE WHEN NumWebVisitsMonth = 0 THEN 0 ELSE ROUND(AVG(NumWebPurchases * 1.0), 2) END AS AvgWebPurchases,

    CASE WHEN NumWebVisitsMonth = 0 THEN 0 ELSE ROUND(AVG(NumWebPurchases * 1.0) / NULLIF(NumWebVisitsMonth, 0), 3) END AS PurchasePerVisit,

    CASE WHEN NumWebVisitsMonth = 0 THEN 0 ELSE ROUND(AVG(MntWines + MntFruits + MntMeatProducts +
        MntFishProducts + MntSweetProducts + MntGoldProds) * 1.0, 0) END AS AvgTotalSpend

FROM marketing_data
GROUP BY NumWebVisitsMonth
ORDER BY NumWebVisitsMonth;

(16 rows affected)

WebVisits | Customers | AvgWebPurchases | PurchasePerVisit | AvgTotalSpend
----------+-----------+-----------------+------------------+--------------
0         | 11        | 0.000000        | 0.000000         | 0.0          
1         | 152       | 3.640000        | 3.645000         | 1238.0       
2         | 202       | 4.220000        | 2.111000         | 1186.0       
3         | 205       | 4.510000        | 1.504000         | 973.0        
4         | 217       | 4.510000        | 1.127000         | 653.0        
5         | 280       | 4.330000        | 0.865000         | 522.0        
6         | 339       | 4.270000        | 0.712000         | 486.0        
7         | 393       | 3.640000        | 0.520000         | 297.0        
8         | 342       | 3.890000        | 0.486000         | 353.0        
9         | 83        | 4.290000        | 0.477000         | 288.0        
10        | 3         | 5.330000        | 0.533000         | 211.0        
13   

<b>
Interestingly, customers who visit the website very frequently (6-8+ times/month) do NOT necessarily make more web purchases, and their average total spend drops sharply (from $1,238 at 1 visit to $297 at 7 visits). The PurchasePerVisit ratio declines from 3.65 to below 0.5, suggesting heavy browsers are comparing prices or abandoning carts without converting. Notably, customers with 0 web visits still spend an average of $1,031, indicating a strong offline/catalog buyer segment. This is a signal for A/B testing personalized offers for heavy browsers and exit-intent campaigns.
<b>

<b><b>

## 4. Customer Lifecycle Analysis

### 4.1 New Customer Enrollment by Month/Year

In [ ]:
-- Customer acquisition trends

SELECT
    YEAR(Dt_Customer) AS EnrollYear,
    MONTH(Dt_Customer) AS EnrollMonth,
    DATENAME(MONTH, Dt_Customer) AS MonthName,
    COUNT(*) AS NewCustomers,

    SUM(COUNT(*)) OVER (ORDER BY YEAR(Dt_Customer), MONTH(Dt_Customer)
                        ROWS UNBOUNDED PRECEDING) AS CumulativeCustomers
                        
FROM marketing_data
GROUP BY
    YEAR(Dt_Customer),
    MONTH(Dt_Customer),
    DATENAME(MONTH, Dt_Customer)
ORDER BY
    EnrollYear,
    EnrollMonth;

(24 rows affected)

EnrollYear | EnrollMonth | MonthName | NewCustomers | CumulativeCustomers
-----------+-------------+-----------+--------------+--------------------
2012       | 7           | July      | 2            | 2                  
2012       | 8           | August    | 118          | 120                
2012       | 9           | September | 99           | 219                
2012       | 10          | October   | 95           | 314                
2012       | 11          | November  | 97           | 411                
2012       | 12          | December  | 83           | 494                
2013       | 1           | January   | 107          | 601                
2013       | 2           | February  | 100          | 701                
2013       | 3           | March     | 102          | 803                
2013       | 4           | April     | 88           | 891                
2013       | 5           | May       | 104          | 995                
2013       | 6    

<b>
The enrollment data spans July 2012–June 2014, totaling 2,236 customers. July 2012 shows only 2 enrollments, likely representing the business launch or first month of data. 2013 was the most consistent acquisition year (88–117 customers/month). Enrollment toward the end of the period shows notable volatility, peaking at 111 in May 2014 but dropping to 74 in June 2014, the lowest month in the dataset, suggesting the acquisition funnel warrants attention. Months with high spikes (Aug 2012: 118, Oct 2013: 117) may align with promotional events. The window function (SUM OVER) provides a running total without requiring a self-join.
<b>

<b>

### 4.2 Customer Tenure vs. Total Lifetime Spend

In [147]:
-- Tenure-based value analysis

WITH Tenure AS (
    SELECT
        ID,
        DATEDIFF(DAY, Dt_Customer, '2015-01-01') AS TenureDays,
        MntWines + MntFruits + MntMeatProducts +
        MntFishProducts + MntSweetProducts + MntGoldProds AS TotalSpend,
        CASE
            WHEN DATEDIFF(DAY, Dt_Customer, '2015-01-01') < 365 THEN '< 1 Year'
            WHEN DATEDIFF(DAY, Dt_Customer, '2015-01-01') < 730 THEN '1-2 Years'
            WHEN DATEDIFF(DAY, Dt_Customer, '2015-01-01') < 1095 THEN '2-3 Years'
            ELSE '3+ Years'
        END AS TenureBand
    FROM marketing_data
)
SELECT
    TenureBand,
    COUNT(*) AS Customers,
    ROUND(AVG(TenureDays), 0) AS AvgTenureDays,
    ROUND(AVG(TotalSpend), 0) AS AvgLifetimeSpend,
    ROUND(MIN(TotalSpend), 0) AS MinSpend,
    ROUND(MAX(TotalSpend), 0) AS MaxSpend
FROM Tenure
GROUP BY TenureBand
ORDER BY AvgTenureDays;

(3 rows affected)

TenureBand | Customers | AvgTenureDays | AvgLifetimeSpend | MinSpend | MaxSpend
-----------+-----------+---------------+------------------+----------+---------
< 1 Year   | 553       | 274           | 492              | 6        | 2525    
1-2 Years  | 1185      | 549           | 602              | 5        | 2524    
2-3 Years  | 498       | 811           | 741              | 13       | 2352    
(3 rows)

Total execution time: 00:00:45.341

<b>
Consistent upward trend: <1 Year ($492) → 1-2 Years ($602) → 2-3 Years ($741). Customers with 2-3 years tenure spend 51% more per customer than new customers, a strong justification for loyalty program investment. While 2-3 Years has the highest average spend per customer ($741), the 1-2 Years cohort generates the most total revenue (~$713K) due to its larger size (1,185 customers). Both segments are priority targets, the former for per-customer value, the latter for volume.
<b>

<b><b>

## 5. Additional Analyses

### 5.1  Country-Level Performance

In [ ]:
SELECT
    Country,
    COUNT(*) AS Customers,
    CAST(ROUND(AVG(Income), 0) AS INTEGER)  AS AvgIncome,

    ROUND(AVG(MntWines + MntFruits + MntMeatProducts +
        MntFishProducts + MntSweetProducts + MntGoldProds) * 1.0, 0) AS AvgTotalSpend,

    CAST(ROUND(AVG(NumWebPurchases * 1.0), 2) AS REAL)     AS AvgWebPurchases,
    CAST(ROUND(AVG(NumStorePurchases * 1.0), 2) AS REAL)   AS AvgStorePurchases,
    ROUND(100.0 * AVG(CAST(Response AS FLOAT)), 1)         AS LastCampaign_AcceptRate,
    ROUND(100.0 * AVG(CAST(Complain AS FLOAT)), 2)         AS ComplaintRate_Pct,

    RANK() OVER (ORDER BY AVG(MntWines+MntFruits+MntMeatProducts+
        MntFishProducts+MntSweetProducts+MntGoldProds) DESC) AS SpendRank

FROM marketing_data
WHERE Income IS NOT NULL
GROUP BY Country
HAVING COUNT(*) >= 10  -- Exclude micro-markets (Mexico: 3 customers)
ORDER BY AvgTotalSpend DESC;

(7 rows affected)

Country      | Customers | AvgIncome | AvgTotalSpend | AvgWebPurchases | AvgStorePurchases | LastCampaign_AcceptRate | ComplaintRate_Pct | SpendRank
-------------+-----------+-----------+---------------+-----------------+-------------------+-------------------------+-------------------+----------
USA          | 107       | 53218     | 631.0         | 4.48            | 6.07              | 12.1                    | 0                 | 1        
Germany      | 116       | 52951     | 631.0         | 4               | 6.03              | 14.7                    | 0.86              | 1        
Saudi Arabia | 335       | 52988     | 629.0         | 4.16            | 5.92              | 15.5                    | 0.9               | 3        
Canada       | 266       | 53051     | 629.0         | 4.29            | 5.84              | 14.3                    | 0.75              | 3        
Spain        | 1092      | 51535     | 602.0         | 4               | 5.79          


USA and Germany jointly lead all markets in average total spend ($631), followed closely by Saudi Arabia and Canada (both $629) — four markets remarkably close in per-customer value despite very different volumes. Among this top tier, Saudi Arabia has the highest campaign acceptance rate at 15.5%, making it the most responsive high-value market.

Spain, ranking 5th in average spend ($602), deserves special attention: it is the largest market by far with 1,092 customers — nearly half the dataset — and holds the highest campaign acceptance rate overall at 16.1%. This combination of scale and responsiveness makes Spain uniquely powerful for broad campaigns.

At the lower end, Australia ($582) and India ($532) show the lowest average spend, suggesting lower purchasing power or a different product-market fit. USA and Australia both show 0% complaint rate — USA alongside a top-tier spend of $631, a strong signal of customer satisfaction.


<b>

### 5.2  Multi-Campaign Responders — VIP Loyalist Profile

In [ ]:
-- 5.2 Multi-Campaign Responder Profile 

WITH CampaignEngagement AS (
    SELECT
        ID,
        CAST(2015 - Year_Birth AS INT)                    AS Age,

        Education,
        Marital_Status,
        Income,
        Kidhome + Teenhome                                AS TotalChildren,

        AcceptedCmp1 + AcceptedCmp2 + AcceptedCmp3 +
        AcceptedCmp4 + AcceptedCmp5 + Response            AS TotalCampaignsAccepted,

        MntWines + MntFruits + MntMeatProducts +
        MntFishProducts + MntSweetProducts + MntGoldProds AS TotalSpend,

        NumWebPurchases + NumCatalogPurchases + 
        NumStorePurchases                                 AS TotalPurchases

    FROM marketing_data
    WHERE Income IS NOT NULL 
      
)

SELECT

    TotalCampaignsAccepted          AS CampaignsAccepted,
    COUNT(*)                        AS Customers,
    CAST(AVG(Age * 1.0) AS INT)     AS AvgAge,
    CAST(ROUND(AVG(Income), 0)      AS INTEGER)  AS AvgIncome,
    ROUND(AVG(TotalSpend), 0)       AS AvgTotalSpend,
    CAST(AVG(TotalPurchases * 1.0)  AS DECIMAL(5,1)) AS AvgPurchases,
    CAST(AVG(TotalChildren * 1.0)   AS DECIMAL(5,1)) AS AvgChildren

FROM CampaignEngagement
GROUP BY TotalCampaignsAccepted
ORDER BY TotalCampaignsAccepted DESC;

(6 rows affected)

CampaignsAccepted | Customers | AvgAge | AvgIncome | AvgTotalSpend | AvgPurchases | AvgChildren
------------------+-----------+--------+-----------+---------------+--------------+------------
5                 | 10        | 47     | 86434     | 1514          | 20.3         | 0.2        
4                 | 36        | 43     | 82143     | 1673          | 20.3         | 0.2        
3                 | 51        | 43     | 78462     | 1486          | 19.5         | 0.3        
2                 | 139       | 46     | 61659     | 1059          | 15.7         | 0.6        
1                 | 368       | 47     | 56751     | 825           | 14.8         | 0.8        
0                 | 1608      | 45     | 48293     | 460           | 11.3         | 1.0        
(6 rows)

Total execution time: 00:00:00.993

<b>
Customers who accepted 4 or more campaigns are the rarest segment (46 customers total) and most valuable, their average spend ($1,514–$1,673) is roughly 3–4x higher than non-responders ($460). These VIP loyalists tend to be higher-income ($82K–$86K vs $48K avg) and nearly child-free (avg 0.2 children vs 1.0). Age is not a distinguishing factor, the 4-campaign group (avg 43) is actually slightly younger than the general population. This profile is the ideal lookalike audience for paid acquisition campaigns. These customers should receive a dedicated loyalty track: early access to new products, exclusive discounts, and personal account management.
<b>

<b><b>

## 6. Business Questions

### BQ1: Factors Correlated with Web Purchases

In [ ]:
-- Factor 1: Income vs. Web Purchases

SELECT
    CASE
        WHEN Income < 40000 THEN 'Low (<$40K)'
        WHEN Income < 70000 THEN 'Mid ($40K-$70K)'
        ELSE 'High ($70K+)'
    END AS IncomeTier,
    CAST(ROUND(AVG(NumWebPurchases * 1.0), 2) AS REAL) AS AvgWebPurchases,
    COUNT(*) AS Customers
FROM marketing_data
WHERE Income IS NOT NULL 
GROUP BY CASE
    WHEN Income < 40000 THEN 'Low (<$40K)'
    WHEN Income < 70000 THEN 'Mid ($40K-$70K)'
    ELSE 'High ($70K+)'
END
ORDER BY AvgWebPurchases DESC;

(3 rows affected)

IncomeTier      | AvgWebPurchases | Customers
----------------+-----------------+----------
High ($70K+)    | 5.26            | 506      
Mid ($40K-$70K) | 4.94            | 975      
Low (<$40K)     | 2.15            | 731      
(3 rows)

Total execution time: 00:00:00.062

<b>
High-income customers purchase 2.4x more via web than low-income customers (5.26 vs 2.15). The gap between High and Mid tiers is modest (0.32), but the drop to Low is dramatic (2.79 purchases), suggesting a purchasing power threshold below $40K where web channel usage falls sharply.
<b>

<b>

In [ ]:

-- Factor 2: Children vs. Web Purchases

SELECT
    Kidhome + Teenhome AS TotalChildren,
    CAST(ROUND(AVG(NumWebPurchases * 1.0), 2) AS REAL) AS AvgWebPurchases,
    COUNT(*) AS Customers
FROM marketing_data
GROUP BY Kidhome + Teenhome
ORDER BY TotalChildren;


(4 rows affected)

TotalChildren | AvgWebPurchases | Customers
--------------+-----------------+----------
0             | 4.39            | 637      
1             | 4.29            | 1125     
2             | 3.23            | 421      
3             | 2.87            | 53       
(4 rows)

Total execution time: 00:00:00.033

<b>
The range across groups is 1.52 purchases, moderate compared to income (3.11) but consistent. Child-free customers purchase the most online, likely due to greater disposable income and time flexibility. The sharpest drop occurs between 1 and 2 children (4.29 → 3.23).
<b>

<b>

In [9]:

-- Factor 3: Education vs. Web Purchases
SELECT
    Education,
    CAST(ROUND(AVG(NumWebPurchases * 1.0), 2) AS REAL) AS AvgWebPurchases,
    CAST(ROUND(AVG(NumWebVisitsMonth * 1.0), 2) AS REAL) AS AvgWebVisits,
    COUNT(*) AS Customers
FROM marketing_data
GROUP BY Education
ORDER BY AvgWebPurchases DESC;

(5 rows affected)

Education  | AvgWebPurchases | AvgWebVisits | Customers
-----------+-----------------+--------------+----------
PhD        | 4.42            | 5.27         | 485      
Graduation | 4.13            | 5.29         | 1126     
Master     | 4.03            | 5.18         | 370      
2n Cycle   | 3.75            | 5.46         | 201      
Basic      | 1.89            | 6.87         | 54       
(5 rows)

Total execution time: 00:00:23.886

<b>
PhD holders lead with 4.42 average purchases. However, the most notable finding is the Basic education anomaly: these customers visit the website the most frequently (6.87 visits/month) yet make the fewest purchases (1.89). This pattern, high browse rate, low conversion  suggests price sensitivity, cart abandonment, or a product-market fit mismatch for this segment. The education purchase range spans 2.53 (PhD to Basic).
<b>

<b>

### BQ2: Campaign Success — Rate + Responder Quality

In [10]:
-- Detailed campaign performance metrics

SELECT 'Campaign 1' AS Campaign, SUM(AcceptedCmp1) AS Accepted,
    ROUND(100.0*AVG(CAST(AcceptedCmp1 AS FLOAT)),1) AS AcceptRate_Pct,

    ROUND(AVG(CASE WHEN AcceptedCmp1=1 THEN
        MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
        AS AvgSpend_Responders,

    ROUND(AVG(CASE WHEN AcceptedCmp1=0 THEN
        MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
        AS AvgSpend_NonResponders

FROM marketing_data

UNION ALL

SELECT 'Campaign 2', SUM(AcceptedCmp2), ROUND(100.0*AVG(CAST(AcceptedCmp2 AS FLOAT)),1),
    ROUND(AVG(CASE WHEN AcceptedCmp2=1 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0),
    ROUND(AVG(CASE WHEN AcceptedCmp2=0 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
FROM marketing_data

UNION ALL

SELECT 'Campaign 3', SUM(AcceptedCmp3), ROUND(100.0*AVG(CAST(AcceptedCmp3 AS FLOAT)),1),
    ROUND(AVG(CASE WHEN AcceptedCmp3=1 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0),
    ROUND(AVG(CASE WHEN AcceptedCmp3=0 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
FROM marketing_data

UNION ALL

SELECT 'Campaign 4', SUM(AcceptedCmp4), ROUND(100.0*AVG(CAST(AcceptedCmp4 AS FLOAT)),1),
    ROUND(AVG(CASE WHEN AcceptedCmp4=1 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0),
    ROUND(AVG(CASE WHEN AcceptedCmp4=0 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
FROM marketing_data

UNION ALL

SELECT 'Campaign 5', SUM(AcceptedCmp5), ROUND(100.0*AVG(CAST(AcceptedCmp5 AS FLOAT)),1),
    ROUND(AVG(CASE WHEN AcceptedCmp5=1 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0),
    ROUND(AVG(CASE WHEN AcceptedCmp5=0 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
FROM marketing_data

UNION ALL

SELECT 'Final (Response)', SUM(Response), ROUND(100.0*AVG(CAST(Response AS FLOAT)),1),
    ROUND(AVG(CASE WHEN Response=1 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0),
    ROUND(AVG(CASE WHEN Response=0 THEN MntWines+MntFruits+MntMeatProducts+MntFishProducts+MntSweetProducts+MntGoldProds END),0)
FROM marketing_data

ORDER BY AcceptRate_Pct DESC;

(6 rows affected)

Campaign         | Accepted | AcceptRate_Pct | AvgSpend_Responders | AvgSpend_NonResponders
-----------------+----------+----------------+---------------------+-----------------------
Final (Response) | 334      | 14.9           | 987                 | 539                   
Campaign 4       | 167      | 7.5            | 1143                | 562                   
Campaign 3       | 163      | 7.3            | 720                 | 596                   
Campaign 5       | 162      | 7.2            | 1614                | 527                   
Campaign 1       | 144      | 6.4            | 1482                | 545                   
Campaign 2       | 30       | 1.3            | 1307                | 596                   
(6 rows)

Total execution time: 00:00:19.787

<b>
The final campaign (Response) leads decisively on acceptance rate (14.9%), more than double any individual campaign, but ranks 5th out of 6 in responder spend quality ($987). The real quality winner is Campaign 5, whose responders average $1,614 in spend with the highest lift ratio (3.06x vs non-responders). Campaign 1 follows closely at $1,482 (2.72x lift). Campaign 2, while the lowest in acceptance rate (1.3%), actually attracts high-quality responders ($1,307), its issue is low reach, not low quality. Campaign 3 shows the weakest responder lift (1.21x), suggesting poor targeting precision. Strategic recommendation: the final campaign's targeting model should be retained for volume, while Campaign 5's targeting criteria should be reverse-engineered for premium segment campaigns.
<b>

<b>

### BQ3: Average Customer Profile

In [11]:
-- Typical customer profile

SELECT
    CAST(AVG(2015.0 - Year_Birth) AS INT) AS AvgAge,

    (SELECT TOP 1 Education FROM marketing_data
     GROUP BY Education ORDER BY COUNT(*) DESC) AS MostCommonEducation,

    (SELECT TOP 1 Marital_Status FROM marketing_data
     GROUP BY Marital_Status ORDER BY COUNT(*) DESC) AS MostCommonMaritalStatus,

    ROUND(AVG(Income), 0) AS AvgIncome,
    ROUND(AVG(Kidhome * 1.0), 2) AS AvgKidsAtHome,
    ROUND(AVG(Teenhome * 1.0), 2) AS AvgTeensAtHome,
    ROUND(AVG(Recency * 1.0), 0) AS AvgDaysSinceLastPurchase,

    ROUND(AVG(MntWines+MntFruits+MntMeatProducts+
        MntFishProducts+MntSweetProducts+MntGoldProds)*1.0,0) AS AvgTotalSpend,

    ROUND(AVG(NumWebPurchases * 1.0), 1) AS AvgWebPurchases,
    ROUND(AVG(NumStorePurchases * 1.0), 1) AS AvgStorePurchases,
    ROUND(AVG(NumCatalogPurchases * 1.0), 1) AS AvgCatalogPurchases,
    ROUND(AVG(NumWebVisitsMonth * 1.0), 1) AS AvgWebVisitsMonth,

    (SELECT TOP 1 Country FROM marketing_data
     GROUP BY Country ORDER BY COUNT(*) DESC) AS MostCommonCountry
     
FROM marketing_data
WHERE Income IS NOT NULL 

(1 row affected)

AvgAge | MostCommonEducation | MostCommonMaritalStatus | AvgIncome    | AvgKidsAtHome | AvgTeensAtHome | AvgDaysSinceLastPurchase | AvgTotalSpend | AvgWebPurchases | AvgStorePurchases | AvgCatalogPurchases | AvgWebVisitsMonth | MostCommonCountry
-------+---------------------+-------------------------+--------------+---------------+----------------+--------------------------+---------------+-----------------+-------------------+---------------------+-------------------+------------------
46     | Graduation          | Married                 | 51959.000000 | 0.440000      | 0.510000       | 49.000000                | 607.0         | 4.100000        | 5.800000          | 2.700000            | 5.300000          | Spain            
(1 row)

Total execution time: 00:00:16.850

<b>
The average customer is approximately 46 years old, holds a university degree (Graduation), is married, earns ~$52K/year, and has roughly 1 child at home (0.44 kids + 0.51 teens). They spend ~$607 annually across products, with the majority on wines (as detailed in Section 2.3). They shop primarily in-store (~6 purchases), followed by web (~4 purchases) and catalog (~3 purchases). They visit the website ~5 times per month and were last active roughly 49 days ago. They are most likely from Spain. This persona should be the benchmark for campaign creative and channel decisions.
<b>

<b>

### BQ4: Product Performance — Revenue & Penetration

In [13]:
-- Product category performance metrics
WITH ProductMetrics AS (
    SELECT
        SUM(MntWines)              AS TotalWines,
        SUM(MntFruits)             AS TotalFruits,
        SUM(MntMeatProducts)       AS TotalMeat,
        SUM(MntFishProducts)       AS TotalFish,
        SUM(MntSweetProducts)      AS TotalSweet,
        SUM(MntGoldProds)          AS TotalGold,

        COUNT(*)                   AS N,

        SUM(CASE WHEN MntWines > 0 THEN 1 ELSE 0 END)          AS BuyersWine,
        SUM(CASE WHEN MntFruits > 0 THEN 1 ELSE 0 END)         AS BuyersFruit,
        SUM(CASE WHEN MntMeatProducts > 0 THEN 1 ELSE 0 END)   AS BuyersMeat,
        SUM(CASE WHEN MntFishProducts > 0 THEN 1 ELSE 0 END)   AS BuyersFish,
        SUM(CASE WHEN MntSweetProducts > 0 THEN 1 ELSE 0 END)  AS BuyersSweet,
        SUM(CASE WHEN MntGoldProds > 0 THEN 1 ELSE 0 END)      AS BuyersGold
    FROM marketing_data
)
SELECT Product, TotalRevenue, PctOfRevenue, BuyerPenetration_Pct
FROM (
    SELECT 'Wines' AS Product,
           TotalWines AS TotalRevenue,
           CAST(ROUND(100.0*TotalWines /(TotalWines+TotalFruits+TotalMeat+TotalFish+TotalSweet+TotalGold),1) AS FLOAT) AS PctOfRevenue,
           CAST(ROUND(100.0*BuyersWine/N, 1) AS FLOAT) AS BuyerPenetration_Pct
    FROM ProductMetrics

    UNION ALL

    SELECT 'Meat', TotalMeat,
           ROUND(100.0*TotalMeat/(TotalWines+TotalFruits+TotalMeat+TotalFish+TotalSweet+TotalGold),1),
           ROUND(100.0*BuyersMeat/N,1)
    FROM ProductMetrics

    UNION ALL

    SELECT 'Gold', TotalGold,
           ROUND(100.0*TotalGold/(TotalWines+TotalFruits+TotalMeat+TotalFish+TotalSweet+TotalGold),1),
           ROUND(100.0*BuyersGold/N,1)
    FROM ProductMetrics

    UNION ALL

    SELECT 'Fish', TotalFish,
           ROUND(100.0*TotalFish/(TotalWines+TotalFruits+TotalMeat+TotalFish+TotalSweet+TotalGold),1),
           ROUND(100.0*BuyersFish/N,1)
    FROM ProductMetrics

    UNION ALL

    SELECT 'Sweet', TotalSweet,
           ROUND(100.0*TotalSweet/(TotalWines+TotalFruits+TotalMeat+TotalFish+TotalSweet+TotalGold),1),
           ROUND(100.0*BuyersSweet/N,1)
    FROM ProductMetrics

    UNION ALL

    SELECT 'Fruits', TotalFruits,
           ROUND(100.0*TotalFruits/(TotalWines+TotalFruits+TotalMeat+TotalFish+TotalSweet+TotalGold),1),
           ROUND(100.0*BuyersFruit/N,1)
    FROM ProductMetrics
) products

ORDER BY TotalRevenue DESC;

(6 rows affected)

Product | TotalRevenue | PctOfRevenue | BuyerPenetration_Pct
--------+--------------+--------------+---------------------
Wines   | 680029       | 50.2         | 99.4                
Meat    | 373375       | 27.6         | 100                 
Gold    | 98346        | 7.3          | 97.3                
Fish    | 83931        | 6.2          | 82.8                
Sweet   | 60552        | 4.5          | 81.3                
Fruits  | 58753        | 4.3          | 82.2                
(6 rows)

Total execution time: 00:00:00.121

<b>
Wines dominate revenue with ~50% share, while Meat leads buyer penetration at 100%, every customer buys meat. Both products are true category anchors. Gold products show high penetration (97.3%) relative to their revenue share (7.3%), suggesting customers buy frequently but in small amounts, a clear upsell opportunity. Fish, Fruits, and Sweets occupy the bottom in both revenue and penetration (81–83%), though even these "niche" products still reach over 80% of customers. Bundling strategies (e.g., "Wine + Chocolate" or "Meat + Fish" gift sets) may help increase basket size for these lower-revenue categories rather than relying on standalone promotion.
<b>

<b>

### BQ5: Channel Performance Analysis

In [14]:
-- Sales channel performance metrics

WITH ChannelStats AS (
    SELECT
        SUM(NumWebPurchases)        AS TotalWebPurch,
        SUM(NumCatalogPurchases)    AS TotalCatalogPurch,
        SUM(NumStorePurchases)      AS TotalStorePurch,
        SUM(NumDealsPurchases)      AS TotalDealsPurch,

        SUM(MntWines+MntFruits+MntMeatProducts+
            MntFishProducts+MntSweetProducts+MntGoldProds) AS TotalSpend,

        SUM(NumWebPurchases+NumCatalogPurchases+NumStorePurchases) AS TotalPurch,
        
        COUNT(*) AS N

    FROM marketing_data
)
SELECT
    Channel,
    PurchaseVolume,
    CAST(ROUND(100.0 * PurchaseVolume / (TotalWebPurch+TotalCatalogPurch+TotalStorePurch+TotalDealsPurch), 1) AS FLOAT) AS VolShare_Pct,
    CAST(ROUND(TotalSpend * 1.0 / NULLIF(PurchaseVolume, 0), 2) AS FLOAT) AS RevenuePerPurchase_Proxy
FROM (
    SELECT 'Web' AS Channel, TotalWebPurch AS PurchaseVolume,
           TotalSpend, TotalWebPurch, TotalCatalogPurch, TotalStorePurch, TotalDealsPurch FROM ChannelStats
    UNION ALL
    SELECT 'Catalog', TotalCatalogPurch, TotalSpend,
           TotalWebPurch, TotalCatalogPurch, TotalStorePurch, TotalDealsPurch FROM ChannelStats
    UNION ALL
    SELECT 'Store', TotalStorePurch, TotalSpend,
           TotalWebPurch, TotalCatalogPurch, TotalStorePurch, TotalDealsPurch FROM ChannelStats
    UNION ALL
    SELECT 'Deals', TotalDealsPurch, TotalSpend,
           TotalWebPurch, TotalCatalogPurch, TotalStorePurch, TotalDealsPurch FROM ChannelStats
) sub
ORDER BY PurchaseVolume DESC;

(4 rows affected)

Channel | PurchaseVolume | VolShare_Pct | RevenuePerPurchase_Proxy
--------+----------------+--------------+-------------------------
Store   | 12959          | 39           | 104.56                  
Web     | 9140           | 27.5         | 148.25                  
Catalog | 5955           | 17.9         | 227.54                  
Deals   | 5201           | 15.6         | 260.52                  
(4 rows)

Total execution time: 00:00:37.246

<b>
Deals is actually the most revenue-efficient channel at $260.52 per purchase, suggesting customers tend to buy in larger quantities when promotions are available. Rather than eroding margins, deals appear to drive higher-value transactions. The real underperformer in efficiency is Store ($104.56 per purchase) despite leading in volume (39% share).
<b>